src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -1,30 +1,71 @@
 package org.breedinginsight.brapi.v2.services;
-
+import io.micronaut.context.annotation.Property;
 import io.micronaut.http.server.exceptions.InternalServerException;
-import org.brapi.client.v2.model.exceptions.ApiException;
+import io.micronaut.http.server.types.files.StreamedFile;
 import lombok.extern.slf4j.Slf4j;
-import org.brapi.v2.model.core.BrAPITrial;
-import org.brapi.v2.model.pheno.BrAPIObservationUnit;
-import org.breedinginsight.brapps.importer.daos.BrAPIObservationUnitDAO;
-import org.breedinginsight.brapps.importer.daos.BrAPITrialDAO;
+import org.apache.commons.lang3.StringUtils;
+import org.brapi.client.v2.model.exceptions.ApiException;
+import org.brapi.v2.model.BrAPIExternalReference;
+import org.brapi.v2.model.core.*;
+import org.brapi.v2.model.core.response.BrAPIListsSingleResponse;
+import org.brapi.v2.model.germ.BrAPIGermplasm;
+import org.brapi.v2.model.pheno.*;
+import org.breedinginsight.brapi.v2.constants.BrAPIAdditionalInfoFields;
+import org.breedinginsight.brapi.v2.dao.BrAPIGermplasmDAO;
+import org.breedinginsight.brapi.v2.model.request.query.ExperimentExportQuery;
+import org.breedinginsight.brapps.importer.daos.*;
+import org.breedinginsight.brapps.importer.model.exports.FileType;
+import org.breedinginsight.brapps.importer.model.imports.experimentObservation.ExperimentObservation;
+import org.breedinginsight.brapps.importer.services.ExternalReferenceSource;
+import org.breedinginsight.model.BrAPIConstants;
+import org.breedinginsight.model.Column;
+import org.breedinginsight.model.DownloadFile;
+import org.breedinginsight.model.Program;
 import org.breedinginsight.services.exceptions.DoesNotExistException;
+import org.breedinginsight.services.parsers.experiment.ExperimentFileColumns;
+import org.breedinginsight.services.writers.CSVWriter;
+import org.breedinginsight.services.writers.ExcelWriter;
 import org.breedinginsight.utilities.Utilities;
-
 import javax.inject.Inject;
 import javax.inject.Singleton;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.*;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Singleton
 public class BrAPITrialService {
-
+    @Property(name = "brapi.server.reference-source")
+    private String referenceSource;
```

Comment:
For testability, this should be included as a parameter to the constructor.  See `BrAPIObservationUnitDAO` as an example

Reply:
done

Reply:
done